data_fetcher.py

```diff
@@ -0,0 +1,45 @@
+import bs4
+import requests
+import lxml
+from bs4 import BeautifulSoup
+from typing import List
+ 
+ 
+class JeuneAfriqueRSSDataSource:
+    
+    def __init__(self, rss_url: str) -> None:
+        self.url = rss_url
+        self.r = None
+        self.status_code = None
+        self.soup = None
+        self.articles = None
+        self.articles_dicts = None
+        self.urls: List[str] = []
+        self.titles: List[str] = []
+        self.descriptions: List[str] = []
+        self.pub_dates: List[str] = []
+        
+    def get_data(self) -> None:
+        try:
+            self.r = requests.get(self.url)
+            self.status_code = self.r.status_code
+        except Exception as e:
```

Comment:
I would recommend not to handle exceptions unless it is unavoidable. It is better when your app fails with an exception, so you can later run pdb and debug what's going on or at least see a stack trace. Otherwise, you are concealing the error. I suggest to read a chapter on exceptions from the book "Code Complete", they give more details there